internal/component/database_observability/mysql/collector/explain_plan.go

```diff
@@ -448,16 +449,32 @@ func parseUnionResultNode(logger log.Logger, unionResultNode []byte) (planNode,
 }
 
 type queryInfo struct {
-	schemaName *string
-	digest     string
-	queryText  string
+	schemaName   string
+	digest       string
+	queryText    string
+	failureCount int
+}
+
+func (qi *queryInfo) key() string {
+	return fmt.Sprintf("%x", md5.Sum([]byte(qi.schemaName+"|"+qi.digest)))
```

Comment:
This method is called multiple times in subsequent code, maybe we could save it in `queryInfo` when it's computed for the first time.

Generally speaking, `sprintf` and concatenations cause a bunch of small allocations that might unnecessarily keep the GC busy. I'm wondering if we need to compute the md5 hash at all... could we just use `qi.schemaName+qi.digest` as cache key? (possibly don't even need the `|` separator, as this is not a piece of debugging information we expose somewhere)